node/pkg/ethereum/watcher.go

```diff
@@ -124,7 +125,16 @@ func NewEthWatcher(
 	messageEvents chan *common.MessagePublication,
 	setEvents chan *common.GuardianSet,
 	minConfirmations uint64,
-	obsvReqC chan *gossipv1.ObservationRequest) *Watcher {
+	obsvReqC chan *gossipv1.ObservationRequest,
+	unsafeDevMode bool) *Watcher {
+
+	var ethIntf common.Ethish
+	if chainID == vaa.ChainIDCelo && ! unsafeDevMode {
+		ethIntf = &celo.CeloImpl{NetworkName: networkName}
+	} else {
+		ethIntf = &EthImpl{NetworkName: networkName}
```

Comment:
Why is this necessary?

It would be best to avoid leaking `unsafeDevMode` into pkg/ and instead have specific flags for the functionality you want to toggle.